packages/expo-apple-authentication/src/AppleAuthentication.types.ts

```diff
@@ -0,0 +1,250 @@
+ * for more details.
+ */
+export type AppleAuthenticationSignInOptions = {
+  user: string;
+
+  /**
+   * The scope of personal information to which your app is requesting access.
+   * The user can choose to deny your app access to any scope at the time of
+   * logging in.
+   * @defaults `[]` (no scopes).
+   */
+  requestedScopes?: AppleAuthenticationScope[];
+
+  /**
+   * Data that’s returned to you unmodified in the corresponding credential
+   * after a successful authentication. Used to verify that the response was
+   * from the request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+};
+
+/**
+ * The options you can supply when making a call to
+ * `AppleAuthentication.refreshAsync()`. You must include the ID string of the
+ * user whose credentials you'd like to refresh.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest)
+ * for more details.
+ */
+export type AppleAuthenticationRefreshOptions = {
+  user: string;
+
+  /**
+   * The scope of personal information to which your app is requesting access.
+   * The user can choose to deny your app access to any scope at the time of
+   * refreshing.
+   * @defaults `[]` (no scopes).
+   */
+  requestedScopes?: AppleAuthenticationScope[];
+
+  /**
+   * Data that’s returned to you unmodified in the corresponding credential
+   * after a successful authentication. Used to verify that the response was
+   * from the request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+};
+
+/**
+ * The options you can supply when making a call to
+ * `AppleAuthentication.logout()`. You must include the ID string of the user to
+ * sign out.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest)
+ * for more details.
+ */
+export type AppleAuthenticationSignOutOptions = {
+  user: string;
+
+  /**
+   * Data that’s returned to you unmodified in the corresponding credential
+   * after a successful authentication. Used to verify that the response was
+   * from the request you made. Can be used to avoid replay attacks.
+   */
+  state?: string;
+};
+
+/**
+ * The user credentials returned from a successful call to
+ * `AppleAuthentication.loginAsync()`, `AppleAuthentication.refreshAsync()`, or
+ * `AppleAuthentication.logoutAsync()`.
+ *
+ * @see [Apple
+ * Documentation](https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidcredential)
+ * for more details.
+ */
+export type AppleAuthenticationCredential = {
+  /**
+   * A value indicating the status type of the requested credential. `SUCCESS`
+   * if the credential was retrieved successfully, or `CANCEL` if the user
+   * canceled the operation before signing in.
+   */
+  type: AppleAuthenticationStatus;
```

Comment:
@tsapeta pointed out that since we don't have a CANCEL status anymore, we can remove this field and enum type.

Reply:
already done :)